### DISCUSSION TITLE
Apps: Can't create installation access token, 404

### DISCUSSION THREAD

**Discussion by Zemnmez** [community/community#71963] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

I feel like this is a bug, but what I'm doing is so simple it can't be.

I'm trying to generate an installation access token for my app, and the endpoint https://api.github.com/app/installations/43225604/access_tokens only ever returns 404 not found. 

I am pretty sure this is the right endpoint (and not the same issue as https://github.com/orgs/community/discussions/24525) because  I'm hitting the endpoint _specified in the JSON response_ of `/app/installations`.

Below, the cURL and response of `/app/installations`:

```
thomas@DESKTOP-B82ERE8 ~> curl -L -H 'Authorization: Bearer [xxx]' -H 'Accept: application/vnd.github+json' --url 'https://api.github.com/app/installations' --header "X-GitHub-Api-V
ersion: 2022-11-28"
[
  {
    "id": 43225604,
    "account": {
      "login": "zemn-me",
      "id": 145876100,
      "node_id": "O_kgDOCLHkhA",
      "avatar_url": "https://avatars.githubusercontent.com/u/145876100?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/zemn-me",
      "html_url": "https://github.com/zemn-me",
      "followers_url": "https://api.github.com/users/zemn-me/followers",
      "following_url": "https://api.github.com/users/zemn-me/following{/other_user}",
      "gists_url": "https://api.github.com/users/zemn-me/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/zemn-me/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/zemn-me/subscriptions",
      "organizations_url": "https://api.github.com/users/zemn-me/orgs",
      "repos_url": "https://api.github.com/users/zemn-me/repos",
      "events_url": "https://api.github.com/users/zemn-me/events{/privacy}",
      "received_events_url": "https://api.github.com/users/zemn-me/received_events",
      "type": "Organization",
      "site_admin": false
    },
    "repository_selection": "selected",
    "access_tokens_url": "https://api.github.com/app/installations/43225604/access_tokens",
```

Below, the cURL and response of the specified endpoint:

```
thomas@DESKTOP-B82ERE8 ~> curl -L -H 'Authorization: Bearer [xxx]' -H 'Accept: application/vnd.github+json' --url 'https://api.github.com/app/installations/43225604/access_tokens' -
-header "X-GitHub-Api-Version: 2022-11-28"
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}
```

**Comment by Zemnmez** [comment, marked as answer]
nevermind, I missed that this is a `POST` endpoint.

**Comment by queenofcorgis** [comment]
We’re glad you could get this solved and we want to make sure others can do the same so we’ve reopened this discussion for better discoverability for our members.

**Reply by ricardogsilva** [reply]
I just stumbled upon this after having made the same mistake as the original poster. I think this should be classified as a bug, as the original error is misleading.

Since the endpoint does in fact exist and the problem is with the request using the wrong HTTP verb, then the response should use a status of `405`, which stands for _Method Not Allowed_ instead of the current `404`, which is _Not Found_  and this should also be the contents of the `message` property of the JSON payload of the response.
